### DISCUSSION TITLE
开发相关问题

### DISCUSSION THREAD

**Discussion by banned-historical-archives** [banned-historical-archives/banned-historical-archives.github.io#365] [Q&A] [answered]
x

**Comment by PirateBook1** [comment]
.git 的文件夹已经超过 1GB 了，关于这方面有什么想法吗

**Reply by PirateBook1** [reply]
@banned-historical-archives

**Reply by banned-historical-archives** [reply]
以后增长应该很慢了，好像也没收到github的警告，先忽略吧

**Comment by PirateBook1** [comment]
@banned-historical-archives 注释的问题，现在的实现恐怕还不太够

比如说这篇：
![image](https://user-images.githubusercontent.com/113572850/194879691-db353e0a-0b28-46bd-bf0d-9dfd145a630b.png)
存在着两种形式的注释，一种叫脚注，一种叫尾注

我在看近几年的国际共运文章时，还看到过注释里面套注释的情况：
![image](https://user-images.githubusercontent.com/113572850/194880249-ef54c120-a638-4a6d-b47a-e891e933417a.png)
尾注里面还有注释，指向其他尾注

有待继续开发

**Reply by banned-historical-archives** [reply]
现在已经没有页码的概念，脚注应该要和尾注合并，这种情况可能下标都需要向后挪1。
注释嵌套不是大问题，5a，9a是怎么回事？

**Reply by banned-historical-archives** [reply]
顶部的1应该是指全文描述，我在其他地方也见过类似的用法。
我们数据库中的注释和描述是分开的，免得有歧义

**Reply by PirateBook1** [reply]
![image](https://user-images.githubusercontent.com/113572850/194882836-d4f612d7-9778-4cd7-ac63-d86683b5071a.png)
5a 照样是尾注

**Reply by banned-historical-archives** [reply]
有些书籍在标题最后用*表示全文描述，在标题中可能还包含数字下标的注释

**Reply by PirateBook1** [reply]
所以我觉得统称注释可能不妥当

**Reply by banned-historical-archives** [reply]
这种标记方式感觉不规范，[5]应该在斯大林的后面，[5a]应该换成[6]

**Reply by PirateBook1** [reply]
总之你看看要不要做兼容吧，只能说是存在这种现象，这份是美革共的上世纪七十年代的文献，他们挺喜欢用这个

**Reply by banned-historical-archives** [reply]
这个兼容起来比较困难，应该很少见吧？

**Reply by PirateBook1** [reply]
正文特定符号框起来的表示这里有注释，框进去的东西就是下面显示的标号。最下面注释部分把正文框起来的标号从上往下显示。我随便说说而已，不知道你目前是怎么实现的

**Reply by banned-historical-archives** [reply]
我存的是整数下标，5a这种不支持的

**Reply by PirateBook1** [reply]
有没有可能换个数据结构开 map 用键值对存

**Reply by banned-historical-archives** [reply]
这个改造成本比较高，校对工具要改，补丁要改。而且会带来很多麻烦，比如以后校对注释的时候要手动输入下标，如果中间插入一个注释，后面的注释的下标全要手动修改

**Reply by PirateBook1** [reply]
代价太大不弄也罢了，序号直接顺延。目前碰到的最复杂的注释形式是四大选集里面那种了。在实践中解决问题

**Comment by ghost** [comment]
請問如何使用mysql連接github建站系統的sql數據庫呢？sci-hub是直接提供sql文件以供下載的，所以可以直接暴力解析，這裡好像不行，因為使用了太多其他編程語言來進行快速解析，所以似乎只能從前端獲得解析結果（就像一個機器學習的AI）。

**Reply by banned-historical-archives** [reply]
https://github.com/banned-historical-archives/banned-historical-archives.github.io/wiki/%E6%96%87%E7%A8%BF%E5%BD%95%E5%85%A5%E3%80%81%E6%A0%A1%E5%AF%B9%E4%B8%8E%E5%BC%80%E5%8F%91%E4%BA%BA%E5%91%98%E8%B4%A1%E7%8C%AE%E4%BB%A3%E7%A0%81

**Reply by banned-historical-archives** [reply]
init-db以后连接本地mysql

**Reply by banned-historical-archives** [reply]
数据库是本地的，不是github托管的

**Reply by banned-historical-archives** [reply]
官网只是一些静态页面，由github actions自动生成

**Reply by ghost** [reply]
> https://github.com/banned-historical-archives/banned-historical-archives.github.io/wiki/%E6%96%87%E7%A8%BF%E5%BD%95%E5%85%A5%E3%80%81%E6%A0%A1%E5%AF%B9%E4%B8%8E%E5%BC%80%E5%8F%91%E4%BA%BA%E5%91%98%E8%B4%A1%E7%8C%AE%E4%BB%A3%E7%A0%81

難道OCR不應該直接使用PaddleOCR的GitHub頁面嗎？PP是Paddle Paddle的意思吧？https://github.com/PaddlePaddle/PaddleOCR

**Reply by ghost** [reply]
> init-db以后连接本地mysql

npm ERR! code 1
npm ERR! path /home/a/banned-historical-archives.github.io
npm ERR! command failed
npm ERR! command sh -c ts-node --project ./tsconfig.build.json ./backend/init-db.ts

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-10-11T02_28_29_784Z-debug.log

**Reply by ghost** [reply]
> init-db以后连接本地mysql

0 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'init-db' ]
1 info using npm@7.5.2
2 info using node@v12.22.12
3 timing config:load:defaults Completed in 3ms
4 timing config:load:file:/usr/share/nodejs/npm/npmrc Completed in 4ms
5 timing config:load:builtin Completed in 4ms
6 timing config:load:cli Completed in 3ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:/home/a/banned-historical-archives.github.io/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/root/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 17ms
19 verbose npm-session 87b6e36fbf8b18dd
20 timing npm:load Completed in 37ms
21 timing command:run-script Completed in 9881ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/usr/share/nodejs/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:314:20)
22 verbose stack     at maybeClose (internal/child_process.js:1022:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
23 verbose pkgid x@0.1.0
24 verbose cwd /home/a/banned-historical-archives.github.io
25 verbose Linux 5.10.0-17-amd64
26 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "init-db"
27 verbose node v12.22.12
28 verbose npm  v7.5.2
29 error code 1
30 error path /home/a/banned-historical-archives.github.io
31 error command failed
32 error command sh -c ts-node --project ./tsconfig.build.json ./backend/init-db.ts
33 verbose exit 1

**Reply by ghost** [reply]
> init-db以后连接本地mysql

無法init-db

**Reply by banned-historical-archives** [reply]
> > https://github.com/banned-historical-archives/banned-historical-archives.github.io/wiki/%E6%96%87%E7%A8%BF%E5%BD%95%E5%85%A5%E3%80%81%E6%A0%A1%E5%AF%B9%E4%B8%8E%E5%BC%80%E5%8F%91%E4%BA%BA%E5%91%98%E8%B4%A1%E7%8C%AE%E4%BB%A3%E7%A0%81
> 
> 難道OCR不應該直接使用PaddleOCR的GitHub頁面嗎？PP是Paddle Paddle的意思吧？https://github.com/PaddlePaddle/PaddleOCR

我没仔细研究，cnocr好像更灵活一些，另外识别速度比pp快不少

**Reply by banned-historical-archives** [reply]
> > init-db以后连接本地mysql
> 
> 無法init-db

node版本有点低，试一试node16?

**Reply by ghost** [reply]
> > > https://github.com/banned-historical-archives/banned-historical-archives.github.io/wiki/%E6%96%87%E7%A8%BF%E5%BD%95%E5%85%A5%E3%80%81%E6%A0%A1%E5%AF%B9%E4%B8%8E%E5%BC%80%E5%8F%91%E4%BA%BA%E5%91%98%E8%B4%A1%E7%8C%AE%E4%BB%A3%E7%A0%81
> > 
> > 
> > 難道OCR不應該直接使用PaddleOCR的GitHub頁面嗎？PP是Paddle Paddle的意思吧？https://github.com/PaddlePaddle/PaddleOCR
> 
> 我没仔细研究，cnocr好像更灵活一些，另外识别速度比pp快不少

PP識別效果應該是最好的，因為其提供了更多的識別結果。

**Reply by banned-historical-archives** [reply]
> > > > https://github.com/banned-historical-archives/banned-historical-archives.github.io/wiki/%E6%96%87%E7%A8%BF%E5%BD%95%E5%85%A5%E3%80%81%E6%A0%A1%E5%AF%B9%E4%B8%8E%E5%BC%80%E5%8F%91%E4%BA%BA%E5%91%98%E8%B4%A1%E7%8C%AE%E4%BB%A3%E7%A0%81
> > > 
> > > 
> > > 難道OCR不應該直接使用PaddleOCR的GitHub頁面嗎？PP是Paddle Paddle的意思吧？https://github.com/PaddlePaddle/PaddleOCR
> > 
> > 
> > 我没仔细研究，cnocr好像更灵活一些，另外识别速度比pp快不少
> 
> PP識別效果應該是最好的，因為其提供了更多的識別結果。

识别用的是pp训练的模型，效果应该是一样的

**Reply by ghost** [reply]
> > > init-db以后连接本地mysql
> > 
> > 
> > 無法init-db
> 
> node版本有点低，试一试node16?

sudo npm run init-db

> x@0.1.0 init-db
> ts-node --project ./tsconfig.build.json ./backend/init-db.ts

Error: ER_ACCESS_DENIED_NO_PASSWORD_ERROR: Access denied for user 'root'@'localhost'
    at Handshake.Sequence._packetToError (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Handshake.ErrorPacket (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/sequences/Handshake.js:123:18)
    at Protocol._parsePacket (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    --------------------
    at Protocol._enqueue (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Protocol.handshake (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/protocol/Protocol.js:51:23)
    at PoolConnection.connect (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/Connection.js:116:18)
    at Pool.getConnection (/home/a/banned-historical-archives.github.io/node_modules/mysql/lib/Pool.js:48:16)
    at /home/a/banned-historical-archives.github.io/node_modules/typeorm/src/driver/mysql/MysqlDriver.ts:1234:18
    at new Promise (<anonymous>)
    at MysqlDriver.createPool (/home/a/banned-historical-archives.github.io/node_modules/typeorm/src/driver/mysql/MysqlDriver.ts:1231:16)
    at MysqlDriver.connect (/home/a/banned-historical-archives.github.io/node_modules/typeorm/src/driver/mysql/MysqlDriver.ts:387:36)
    at DataSource.initialize (/home/a/banned-historical-archives.github.io/node_modules/src/data-source/DataSource.ts:232:27)
    at /home/a/banned-historical-archives.github.io/backend/data-source.ts:49:18 {
  code: 'ER_ACCESS_DENIED_NO_PASSWORD_ERROR',
  errno: 1698,
  sqlMessage: "Access denied for user 'root'@'localhost'",
  sqlState: '28000',
  fatal: true
}

**Reply by banned-historical-archives** [reply]
你搜一下如何给mysql 设置一个密码，可以使用dbeaver 测试一下

**Reply by ghost** [reply]
> 你搜一下如何给mysql 设置一个密码，可以使用dbeaver 测试一下

ERROR 1049 (42000): Unknown database 'root@localhost'

**Reply by ghost** [reply]
dbeaver沒有在我的軟件源內，只能之後下載了。

**Reply by banned-historical-archives** [reply]
创建一个banned_history的数据库

**Reply by ghost** [reply]
> 创建一个banned_history的数据库

sudo mysql -u root -p
Enter password: 
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 55
Server version: 10.5.15-MariaDB-0+deb11u1 Debian 11

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> show databases;
+--------------------+
| Database           |
+--------------------+
| banned_history     |
| information_schema |
| mysql              |
| performance_schema |
+--------------------+
4 rows in set (0.000 sec)

之後npm run init-db，仍然是重複之前的錯誤。

**Reply by ghost** [reply]
> 创建一个banned_history的数据库

https://forums.puri.sm/t/really-no-mysql-server-package-in-pureos/2244

是否和這個有關？

**Reply by banned-historical-archives** [reply]
MariaDB ?

**Reply by PirateBook1** [reply]
@hubred-rev 密码用 root_password

**Reply by banned-historical-archives** [reply]
你看看root密码是不是和wiki中的一致？否则你需要改一下密码或者改一下环境变量

**Reply by banned-historical-archives** [reply]
为啥要用MariaDB

**Reply by banned-historical-archives** [reply]
完整的错误日志发一下？

**Reply by PirateBook1** [reply]
@banned-historical-archives MariaDB 不是问题，我用的也是这个

**Reply by ghost** [reply]
> 完整的错误日志发一下？

看錯了，需要改密碼。

**Reply by ghost** [reply]
> 完整的错误日志发一下？

謝謝各位，我已經完成無錯初始化了。

**Reply by ghost** [reply]
> 完整的错误日志发一下？

運行效果：

“四人帮”罪行材料（四）
parsed “四人帮”罪行材料（四）
done “四人帮”罪行材料（四）
“四人帮”罪行材料（一〇）
parsed “四人帮”罪行材料（一〇）
done “四人帮”罪行材料（一〇）
“四人帮”罪行材料（五）
parsed “四人帮”罪行材料（五）
done “四人帮”罪行材料（五）
王洪文、纪登奎同志在省委工作会议召集人和县委书记会议上的讲话
parsed 王洪文、纪登奎同志在省委工作会议召集人和县委书记会议上的讲话
done 王洪文、纪登奎同志在省委工作会议召集人和县委书记会议上的讲话
王洪文、叶剑英在接见总参民兵组训工作座谈会议全体同志时的讲话
parsed 王洪文、叶剑英在接见总参民兵组训工作座谈会议全体同志时的讲话
done 王洪文、叶剑英在接见总参民兵组训工作座谈会议全体同志时的讲话
毛远新同志在辽联赴京代表团座谈会上的讲话纪要
parsed 毛远新同志在辽联赴京代表团座谈会上的讲话纪要
done 毛远新同志在辽联赴京代表团座谈会上的讲话纪要
坚决批倒批臭刘少奇反革命修正主义路线
parsed 坚决批倒批臭刘少奇反革命修正主义路线
done 坚决批倒批臭刘少奇反革命修正主义路线
毛泽东选集1-7(静火版)

**Reply by PirateBook1** [reply]
npm run init-db 需要大概十几分钟，等就是了

**Reply by PirateBook1** [reply]
解析完成之后你要什么数据直接从数据库里面拿就是了

**Comment by PirateBook1** [comment]
@banned-historical-archives 目前资料库是怎么确定一篇文章的，有唯一确定的 ID 吗。依靠标题+作者+日期的方式归并文章总觉得不太合适

**Reply by ghost** [reply]
> @banned-historical-archives 目前资料库是怎么确定一篇文章的，有唯一确定的 ID 吗。依靠标题+作者+日期的方式归并文章总觉得不太合适

有ID，['0000f567d8', '119caeb7c1', '314ad57c4d', '3bff1d2fc6', '523bb79e8f', '57434e107b', 'c20d333791', 'd48f0ea3e8', 'fa1994871e', 'wenku', '9014e20fdf', 'f7b7498ef1', '000330f256', '15c8b9e203', 'wenji6', '261e5467ec', '000c0ebcf8', '9132f5a2c9', 'd5cbd05390', 'maoquanji51', '0022c11d9e', 'c36e9054b2', '1ffce755c0', '8173a58a97', 'c0a0ee982d', 'c9f57d6426', 'ef6e778547', '00260e5f8d', 'cd0e3abc04', '07ed767d50', '4b53b7c120', 'c1c0033e04', '9fe446bf79', '002661377e', '4c0eca3228', 'yaowenyuan', '45f848c5df', '002ad4f963', '94ebab6c8a', 'wenji2', '0041a16a1f', '2683092f4c', '2d13338754', 'b88b47c25d', '6366fade06', '00524d1660', 'c44dd9c053', '9bc2c334c7', 'maoquanji39', '00540b9d04', '78c06258c0', 'e23dcf99fa', 'wanghongwen9', 'b2b7bbb332', '0057b25b0b', '9cb0ba9fb5', '2bffb53b5b', '2855c5cb0e', '005c53c243', 'f141974508', 'maoquanji35', '00694a768f', '451e1994a2', 'maoquanji50', '00716048cd', '76838473f3', 'maoxuan-jinghuo', 'xuanji5', '0072f8c4c0', 'beb3a6d028', 'maoquanji49', '00768d2d70', '59bc7a3aeb', 'maoquanji37', '00889f1871', '54210e0667', '18bf6a78fe', '714adabb18', 'c0cd6cf1d4', '0094be272e', 'd45d9aba9b', 'wansui3', '0097c80d24', '66d985de8e', 'maoquanji30', '0098584ec2', 'c7ebe13682', 'maoquanji33', '009867345c', '85ce50a0c6', '91736a0628', '0099417368', 'f86d5c6f1c', '009ae20225', 'f5ed2106f5', 'xuanji4', '009f46cf7f', '80a3be1600', 'wansui1', '00a96e54d3', '20a871bf31'這些。

**Reply by banned-historical-archives** [reply]
```
export function get_article_id(r: ParserResult) {
  return hash_str_arr([
    r.title,
    JSON.stringify(
      r.dates.sort((a, b) =>
        `${a.year || '0000'}-${ensure_two_digits(a.month)}-${ensure_two_digits(
          a.day,
        )}` >
        `${b.year || '0000'}-${ensure_two_digits(b.month)}-${ensure_two_digits(
          b.day,
        )}`
          ? 1
          : -1,
      ),
    ),
    JSON.stringify(r.is_range_date),
    JSON.stringify(r.authors.sort((a, b) => (a > b ? 1 : -1))),
    JSON.stringify(r.file_id || ''),
  ]);
}
```

**Reply by banned-historical-archives** [reply]
title + dates + is_range_date + authors + file_id(默认为空，在中央文件中用的上，可能在同一天发出多个标题一致的通知，这时需要用file_id作区分)

**Reply by PirateBook1** [reply]
![image](https://user-images.githubusercontent.com/113572850/194988283-fd1ee95d-9ed4-425a-9624-d13a08ece0db.png)

**Reply by PirateBook1** [reply]
所以说一定要标题、日期、作者全等才能合并吗，总觉得不太合理

**Reply by banned-historical-archives** [reply]
时间不同。

**Reply by banned-historical-archives** [reply]
合理的，有些例子是标题一致，日期一致，作者不一致

**Reply by banned-historical-archives** [reply]
我觉得文库第一版也可以进行根据标题修改一下日期等操作，和第三版的录入不冲突就行（以后替换成第三版没有副作用就行）

**Reply by PirateBook1** [reply]
我觉得可以增加一个人工控制合并的优先级最高的变量。就像毛远新的这篇，它虽然不符合机器的合并要求，但是事实上它就是一篇一模一样的应该合并起来的文稿

合并处理不好的话对资料库的对比功能很有害

**Reply by banned-historical-archives** [reply]
人工控制那和在文库解析器里面写if else也没什么区别吧

**Comment by PirateBook1** [comment]
@banned-historical-archives 书籍你打算是以什么形式保存，目前的标题+日期+作者应该不适合了吧

**Reply by banned-historical-archives** [reply]
还没想好，感觉差不多是这样

**Reply by PirateBook1** [reply]
报刊、杂志和书籍其实还是有很多不同

**Comment by PirateBook1** [comment]
@banned-historical-archives 注释内部可以换行吗

![image](https://user-images.githubusercontent.com/113572850/195013393-d3e72836-b26b-489d-8395-55b7a19a74d4.png)

**Reply by banned-historical-archives** [reply]
应该可以

**Reply by banned-historical-archives** [reply]
但不一定显示换行

**Reply by PirateBook1** [reply]
这样的揭批查文稿全盘尊重原稿排版吧，不折腾了

**Comment by ghost** [comment]
mysqldump導出的SQL文件還是不夠統一標準，不像Sci-Hub的SQL有整齊劃一的特點，此外，我建議把SQL文件作為release，或者把TXT文件作為release發表。

**Reply by ghost** [reply]
如果追求成果的話，還是前端網絡爬蟲比較一次性理清條理。

**Reply by ghost** [reply]
https://github.com/ProletRevDicta/Prolet/tree/master/A%20%E6%95%A3%E8%A3%85%E5%BB%BA%E5%9B%BD%E5%90%8E%E3%80%8A%E6%AF%9B%E6%B3%BD%E4%B8%9C%E5%85%A8%E9%9B%86%E3%80%8B%E3%80%81%E4%B8%AD%E5%9B%BD%E6%8C%81%E4%B9%85%E4%BA%BA%E6%88%98%E3%80%81%E6%97%A0%E4%BA%A7%E9%98%B6%E7%BA%A7%E6%96%87%E5%8C%96%E5%A4%A7%E9%9D%A9%E5%91%BD%E8%B5%84%E6%96%99%E6%B1%87%E7%BC%96%20-%20%E5%92%8C%E8%B0%90%E5%8E%86%E5%8F%B2%E6%A1%A3%E6%A1%88%E9%A6%86%E7%BC%96%202022.10.10%E7%89%88

**Reply by PirateBook1** [reply]
@banned-historical-archives 可以定期发一发，比如说每一个月发布一次

**Reply by ghost** [reply]
> @banned-historical-archives 可以定期发一发，比如说每一个月发布一次

此外，Google的搜索引擎真的不會收錄那麼多GitHub內容，請停用Google的搜索引擎，很多內容都搜索不到。

**Reply by banned-historical-archives** [reply]
不夠統一標準是指？

**Reply by ghost** [reply]
> 不夠統一標準是指？

元數據不是明顯儲存的，元數據獲取困難。另外就是好像是分成兩個大表保存的，不夠統一。

**Reply by banned-historical-archives** [reply]
关系型数据库是这样的

**Reply by banned-historical-archives** [reply]
https://github.com/banned-historical-archives/banned-historical-archives.github.io/tree/master/backend/entity
表的关系看这个比较清晰

**Reply by PirateBook1** [reply]
@banned-historical-archives 把提出来的 .sql 发一下 Release 吧

**Comment by ghost** [comment]
Google的搜索引擎還是移除為好，除非網站能提供超長列表供瀏覽器進行瀏覽器內搜索，不然還是移除搜索功能為好，Google的搜索引擎基本上什麼都搜索不到，Google的鏈接收錄是有限度的，並且還要看網站訪問流量。

**Reply by ghost** [reply]
Prolet至今為止還難以在Google上搜索得到就表明Google實際是帝國主義的壟斷搜索引擎。

**Reply by PirateBook1** [reply]
同意，最好是基于数据库来搜索

**Reply by banned-historical-archives** [reply]
如果使用私人服务器做搜索服务很难保证匿名

**Reply by banned-historical-archives** [reply]
duckduckgo如何？

**Reply by banned-historical-archives** [reply]
duckduckgo不能提交sitemap，忽略

**Reply by banned-historical-archives** [reply]
https://restoreprivacy.com/private-search-engine/

**Reply by banned-historical-archives** [reply]
https://yacy.net 这个似乎不错，但需要客户端（那还不如本地build from source 本地搜索）

**Comment by banned-historical-archives** [comment, marked as answer]
@hubred-rev @PirateBook1 这里有些凌乱，以后具体技术问题直接在discussions 里面开帖子讨论吧
